react/components/NumericStepper/index.js

```diff
@@ -284,9 +284,11 @@ class NumericStepper extends Component {
     } `
 
     const touchDevice =
-      'ontouchstart' in window ||
-      navigator.maxTouchPoints ||
-      navigator.msMaxTouchPoints
+    'ontouchstart' in window
+      ? true
+      : typeof navigator === 'undefined'
+      ? false
+      : navigator?.maxTouchPoints || navigator?.msMaxTouchPoints
```

Comment:
This can be simpler, right? I've put the `navigator != null` just because I'm not 100% sure about the support of the optional chaining. If we transpile it correctly on JS, then it's better to use it.

```suggestion
    'ontouchstart' in window || (navigator != null && (navigator.maxTouchPoints || navigator.msMaxTouchPoints))
```

Reply:
unfortunately that doesn't work. the error occurs when trying to access the value of the variable when it doesn't exist, so I need to use typeof 😞

Reply:
Oh, I see... Maybe this, then? 👇 
```suggestion
    'ontouchstart' in window || (typeof navigator !== 'undefined' && (navigator.maxTouchPoints || navigator.msMaxTouchPoints))
```

Reply:
done!